src/utils/githubApi.ts

```diff
@@ -95,25 +97,27 @@ export function getGithubApiToken(): string {
  * @param token Github authentication token
  * @returns Github client
  */
-export function getGithubClient(token = ''): Octokit {
+export function getGitHubClient(token = ''): Octokit {
   const githubApiToken = token || getGithubApiToken();
 
-  const attrs = {
-    auth: `token ${githubApiToken}`,
-  } as any;
-
-  if (logger.level >= LogLevel.Debug) {
-    attrs.log = {
-      info: (message: string, _: any) => {
-        logger.debug(message);
-      },
+  if (!_GitHubClientCache[githubApiToken]) {
+    const attrs: any = {
+      auth: `token ${githubApiToken}`,
     };
+
+    if (logger.level >= LogLevel.Debug) {
+      attrs.log = {
+        info: (message: string) => logger.debug(message),
+      };
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { retry } = require('@octokit/plugin-retry');
+    const octokitWithRetries = Octokit.plugin(retry);
+    _GitHubClientCache[githubApiToken] = new octokitWithRetries(attrs);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const { retry } = require('@octokit/plugin-retry');
-  const octokitWithRetries = Octokit.plugin(retry);
-  return new octokitWithRetries(attrs);
+  return _GitHubClientCache[githubApiToken];
```

Comment:
Any reason to introduce this cache on this particular PR, or is this more of an opportunistic optimization?

Reply:
During my refactoring, I moved the calls to `getGitHubClient` function into the two helper functions because it felt stupid to call it from outside and keep passing it into the helpers. Turns out we never need a new client throughout the application unless we provide a new token so I made the optimization.

So it is both I guess 🙂